### PR TITLE
Soc 986

### PR DESCRIPTION
**Pull request by qaga**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 1]
```diff
@@ -17,138 +18,150 @@
...
+        return true;
+    }
+
+    public boolean isNavListElementEllipsized(int index) {
+        wait.forElementVisible(navList.get(index));
+        return navList.get(index).getCssValue("text-overflow").equals("ellipsis");
+    }
+
+    public boolean isMenuFieldVisible() {
+        return menuField.getCssValue("visibility").equals("visible");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Move the "visible" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 2]
```diff
@@ -17,138 +18,150 @@
...
+        wait.forElementVisible(randomPageButton);
+        randomPageButton.click();
+
+        return new ArticlePageObject(driver);
+    }
+
+    public boolean isSuggestionListDisplayed() {
+        try {
+            wait.forElementVisible(searchSuggestion, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 3]
```diff
@@ -17,138 +18,150 @@
...
+        wait.forElementVisible(navList.get(index));
+        return navList.get(index).getCssValue("text-overflow").equals("ellipsis");
+    }
+
+    public boolean isMenuFieldVisible() {
+        return menuField.getCssValue("visibility").equals("visible");
+    }
+
+    public boolean isResultFieldVisible() {
+        return resultField.getCssValue("visibility").equals("visible");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Move the "visible" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 4]
```diff
@@ -17,138 +18,150 @@
...
+        } catch (TimeoutException | NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isRandomPageButtonDisplayed() {
+        try {
+            wait.forElementVisible(randomPageButton, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 5]
```diff
@@ -17,138 +18,150 @@
...
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isSorryInfoDisplayed() {
+        try {
+            wait.forElementVisible(sorrySpan, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 6]
```diff
@@ -17,138 +18,150 @@
...
+        if (menuView.getAttribute("class") == null) {
+            throw new WebDriverException("Expected String but got null");
+        }
+        return !menuView.getAttribute("class").contains("collapsed");
+    }
+
+    public boolean isBackLinkDisplayed() {
+        try {
+            wait.forElementVisible(backChevron, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java [hunk 7]
```diff
@@ -17,138 +18,150 @@
...
+            wait.forElementVisible(randomPageButton, 5, 1000);
+        } catch (TimeoutException | NoSuchElementException e) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean isNavListElementEllipsized(int index) {
+        wait.forElementVisible(navList.get(index));
+        return navList.get(index).getCssValue("text-overflow").equals("ellipsis");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/NavigationSideComponentObject.java]: MAJOR Move the "ellipsis" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 7 major

Watch the comments in this conversation to review them.

**Comment by tomasznapieralski** [comment]
I'm wondering why git doesn't show changes correctly, but anyway LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues:
* MAJOR 4 major
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/JoinPageObject.java#L37) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPageObject.java#L119) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPageObject.java#L56) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/CropImageTests.java#L38) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Define a constant instead of duplicating this literal "Curated Content" 4 times.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java#L97) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)
* MAJOR [2 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/a69a85088c3c8a5624d423d2a678d7599fbb54b7/src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/EditorTests.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

**Comment by tomasznapieralski** [comment]
Tests were passing on both local and Jenkins

**Pull request merged**